kernel/kernel-api/src/main/java/io/delta/kernel/Scan.java

```diff
@@ -153,7 +153,7 @@ private void initIfRequired() {
         physicalReadSchema = ScanStateRow.getPhysicalSchema(scanState);
         logicalReadSchema = ScanStateRow.getLogicalSchema(scanState);
 
-        tablePath = ScanStateRow.getTableRoot(scanState);
+        tablePath = ScanStateRow.getTableRoot(scanState).toString();
```

Comment:
I took a look into how this is used downstream, it seems like eventually we do convert it back to a Path. Ideally I think we want to start only using Path in kernel code and not using String except at API boundaries. But no need to fix that in this PR

Reply:
>only using Path in kernel code and not using String except at API boundaries
that's a good call and could be less error-prone. We could work on refactoring later.